CMakeLists.txt

```diff
@@ -123,6 +113,19 @@ include(cmake/secp256k1.cmake)
 include(CheckStdFilesystem)
 check_std_filesystem()
 
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
+  include(CheckPIESupported)
+  check_pie_supported(OUTPUT_VARIABLE check_pie_output LANGUAGES CXX)
+  if(CMAKE_CXX_LINK_PIE_SUPPORTED)
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+  endif()
+else()
+  check_cxx_source_links_with_flags(-fPIE "int main(){}" COMPILER_SUPPORTS_PIE)
```

Comment:
This is missing a few things compared to the autotools check:
`AX_CHECK_LINK_FLAG([-fPIE -pie], [PIE_FLAGS="-fPIE"; HARDENED_LDFLAGS="$HARDENED_LDFLAGS -pie"], [], [$CXXFLAG_WERROR])`

- This needs `-pie` as well to check the link option.

- need to make sure we're checking for warnings, as several platforms spew annoying warnings rather than errors.